addons/account/models/account_move.py

```diff
@@ -2540,6 +2542,30 @@ def _get_quick_edit_suggestions(self):
             self.quick_edit_total_amount - self.amount_total)['total_excluded']
         return {'account_id': account_id, 'tax_ids': taxes.ids, 'price_unit': price_untaxed}
 
+    @api.onchange('quick_edit_mode', 'journal_id', 'company_id')
+    def _quick_edit_mode_suggest_invoice_date(self):
+        for record in self:
+            if record.invoice_date:
+                continue
+            if record.quick_edit_mode:
+                # Suggest the Customer Invoice/Vendor Bill date based on previous invoice and lock dates
+                invoice_date = date.today()
+                prev_move = self.search([('state', '=', 'posted'),
+                                         ('journal_id', '=', record.journal_id.id),
+                                         ('company_id', '=', record.company_id.id),
+                                         ('invoice_date', '!=', False)],
+                                        limit=1)
+                if prev_move:
+                    lock_date = prev_move.company_id._get_user_fiscal_lock_date()
+                    prev_date_last_day_of_month = prev_move.invoice_date + relativedelta(day=31)
+                    if lock_date and prev_date_last_day_of_month <= lock_date:
+                        invoice_date = lock_date + relativedelta(months=1) + relativedelta(day=31)
+                    elif prev_date_last_day_of_month < date.today():
+                        invoice_date = prev_date_last_day_of_month
+                record.invoice_date = invoice_date
+            else:
+                record.invoice_date = False
```

Comment:
This is useless since we already asserted that `not record.invoice_date` is true.

So the only condition could be
```python
    if record.quick_edit_mode and not record.invoice_date:
        # Suggest the Customer Invoice/Vendor Bill date based on previous invoice and lock dates
        ...